src/lib.rs

```diff
@@ -146,22 +173,25 @@ impl Watchers {
         }
 
         let entry = self.getter_map.entry(tax_id).or_insert(Vec::new());
-        entry.push(Arc::downgrade(&watcher));
+        entry.push((range, Arc::downgrade(&watcher)));
 
         WatcherGuard {
             key: index,
             map: self.map.clone()
         }
     }
 
-    fn get(&self, index: usize) -> Option<Arc<SyncExtSender>> {
+    fn get(&self, index: usize) -> Option<Arc<SyncSender>> {
```

Comment:
I would suggest wrapping these `usize` as a less fragile type. You can do it in a followup, though.